src/components/Visualization/useAnalyticsData.js

```diff
@@ -43,8 +43,21 @@ const excludedDimensions = [
     DIMENSION_ID_LAST_UPDATED_BY,
 ]
 
-const findOptionSetItem = (code, metaDataItems) =>
-    Object.values(metaDataItems).find((item) => item.code === code)
+const findOptionSetItem = (optionSetId, code, metaDataItems) => {
+    const optionSetMetaData = metaDataItems[optionSetId]
```

Comment:
The only way I could break this function was by passing null or undefined for `metaDataItems` (or leave it out). Maybe do
`const optionSetMetaData = metaDataItems?.[optionSetId]`
to avoid that.

Reply:
Done.